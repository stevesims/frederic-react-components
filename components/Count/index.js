import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

const StyledCount = styled('div')({
  fontFamily: 'Roboto, Arial, sans-serif',
});

const Count = props => {
  const { value } = props;

  return (
    <StyledCount>
      <span>{value}</span>
    </StyledCount>
  );
};

Count.propTypes = {
  textClass: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
};

export default Count;
