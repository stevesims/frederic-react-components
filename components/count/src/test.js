import React from 'react';
import { mount, shallow } from 'enzyme';

import Count from '.';

describe('Count', () => {
  it('renders with only required props', () => {
    const wrapper = shallow(<Count value={2} />);
    expect(wrapper.exists()).toBe(true);
  });

  it('displays number passed in correctly', () => {
    const wrapper = shallow(<Count value={4} />);
    expect(wrapper.find('span').text()).toBe('4');
  });

  it('matches snapshot', () => {
    const wrapper = mount(<Count value={6} />);
    expect(wrapper).toMatchSnapshot();
  });
});
