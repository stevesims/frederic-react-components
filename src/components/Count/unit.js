import test from 'tape';
import React from 'react';
import { shallow } from 'enzyme';

import Count from '.';

test(
  'Component - Count renders',
  assert => {
    shallow(<Count/>);
    assert.end();
  }
);

test(
  'Component - Count displays number passed in correctly',
  assert => {
    const component = shallow(<Count value={4}/>);
    assert.equal(component.find('span').text(), '4');
    assert.end();
  }
);
