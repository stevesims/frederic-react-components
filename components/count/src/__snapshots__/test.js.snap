// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Count matches snapshot 1`] = `
<StyledCount>
  <span>
    4
  </span>
</StyledCount>
`;
